src/lib/ngsi/EntityIdVector.cpp

```diff
@@ -120,6 +120,41 @@ void EntityIdVector::present(const std::string& indent)
 
 /* ****************************************************************************
 *
+* EntityIdVector::lookup - find a matching entity in the entity-vector
+*/
+EntityId* EntityIdVector::lookup(const std::string& id, const std::string& type, const std::string& isPattern)
+{
+  //
+  // isPattern:  "false" or "" is the same
+  //
+  std::string isPatternFromParam = isPattern;
+  if (isPatternFromParam == "")
+  {
+    isPatternFromParam = "false";
+  }
+
+  for (unsigned int ix = 0; ix < vec.size(); ++ix)
+  {
+    std::string isPatternFromVec = vec[ix]->isPattern;
+
+    if (isPatternFromVec == "")
+    {
+      isPatternFromVec = "false";
+    }
+
+    if ((vec[ix]->id == id) && (vec[ix]->type == type) && (isPatternFromVec == isPatternFromParam))
```

Comment:
I looked into this a bit closer and I added a comment for the EntityIdVector struct, as you will see in commit e3b872e.

I propose to use the method-name 'lookup' to lookup exact matches and 'match' to find matching entities.
I was thinking about calling it lookupExact to avoid confusion but I don't like the name very much ...

If we can agree on using 'match' for matching functions/methods and lookup for **exact** searches, then we are all good ...


What I **really** would need here is a method to compare which EntityId is 'broader' (which one 'eats' the other) and replace the 'smaller' with the 'bigger'. However, this is a pretty complex task and including both entities for the forwarded query is 100% OK - just not as elegant.

Imagine we have a query to be forwarded with { E.*, A1 } and { E3, A2 }.
With the current implementation, the forwarded query will be:  { [ E.*, E3 ], [ A1, A3 ] }.
As E.* 'eats' E3, it would be more elegant to forward { [ E.* ], [ A1, A2 ] }.

Not worth the effort ...

So, with a comment, fixed in e3b872e